low-level/framework/utils/salt_util.py

```diff
@@ -13,88 +13,114 @@
+    def get_node_id():
+        '''Returns salt minion_id using salt config file'''
+
+        _node_id = 'srvnode-1'
         try:
-            subout = subprocess.Popen(self.GRAINS_GET_NODE_CMD, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-            result = subout.stdout.readlines()
-            if result == [] or result == "":
-                logger.warning(f"Command '{self.GRAINS_GET_NODE_CMD}' failed to fetch grain id or hostname.")
-                subout = subprocess.Popen(self.MINION_GET_NODE_CMD, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-                result = subout.stdout.readlines()
-                if result == [] or result == "":
-                    logger.warning(f"Command '{self.MINION_GET_NODE_CMD}' failed to fetch minion id or hostname.")
-                    logger.warning("Using node_id ('srvnode-1') as we are not able to fetch it from hostname command.")
-                    node_key = 'srvnode-1'
-                else:
-                    node_key = result[0].decode().rstrip('\n').lower()
-            else:
-                node_key = result[0].decode().rstrip('\n')
-        except Exception as e:
-            logger.warning(f"Can't read node id, using 'srvnode-1' : {e}")
-        return node_key
+            with open('/etc/salt/minion_id', 'r') as salt_file:
+                _node_id = salt_file.read()
+            if _node_id is not None:
+                _node_id = 'srvnode-1'
+        except Exception as err:
+            logger.warning(f'sspl_constants, Fail to read node-id with error: {err}, \
+                             Keeping default value as srvnode-1')
+            _node_id = 'srvnode-1'
+        return _node_id
 
-    def is_single_node(self):
+    def _is_single_node(self):
         """
         Returns true if single node, false otherwise
         """
-        is_single_node = True
+        _is_single_node = False
         try:
-            SALT_CMD = "sudo salt-call pillar.get cluster:type --output=newline_values_only"
-            subout = subprocess.Popen(SALT_CMD, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-            result = subout.stdout.readlines()
-            if result == [] or result == "":
-                logger.warning("Cluster type fetch failed, assuming single node setup.")
+            if self.pillar_info:
+                cluster_type = \
+                    self.pillar_info[self.node_id][self.CLUSTER_KEY][self.CLUSTER_TYPE_KEY]
+                if cluster_type is not None and cluster_type.lower() == 'single':
+                    _is_single_node = True
             else:
-                if result[0].decode().rstrip('\n') != "single":
-                    is_single_node = False
-        except Exception as e:
-            logger.warning(f"Cluster type read failed, assuming single node setup : {e}")
-        return is_single_node
+                logger.warning(f'sspl_constants, Fail to read cluster type with \
```

Comment:
Where is _err set in this function ? Should it be self._err ? 
Did you test else scenario ?

Reply:
Now I have fixed this. Tested else scenario as well.